crates/iceberg/src/spec/name_mapping/mod.rs

```diff
@@ -179,44 +393,48 @@ mod tests {
     #[test]
     fn test_json_name_mapping_deserialization() {
         let name_mapping = r#"
-        [
             {
-                "field-id": 1,
-                "names": [
-                    "id",
-                    "record_id"
-                ]
-            },
-            {
-                "field-id": 2,
-                "names": [
-                    "data"
-                ]
-            },
-            {
-                "field-id": 3,
-                "names": [
-                    "location"
-                ],
-                "fields": [
+                "root": [
```

Comment:
From what I understood, this JSON shouldn't change (`root` field and inclusion of the mappings below), which can be achieved using the `serde(skip)` derive too.

Although, I'll defer to someone more with a lot more experience of the iceberg project too here :smile: